Mathlib/LinearAlgebra/BilinearForm/Hom.lean

```diff
@@ -199,114 +191,104 @@ namespace LinearMap
 variable {R' : Type*} [CommSemiring R'] [Algebra R' R] [Module R' M] [IsScalarTower R' R M]
 
 /-- Apply a linear map on the output of a bilinear form. -/
-@[simps]
-def compBilinForm (f : R →ₗ[R'] R') (B : BilinForm R M) : BilinForm R' M where
-  bilin x y := f (B x y)
-  bilin_add_left x y z := by simp only [BilinForm.add_left, map_add]
-  bilin_smul_left r x y := by
-    simp only
-    rw [← smul_one_smul R r (_ : M), BilinForm.smul_left, smul_one_mul r (_ : R), map_smul,
-      smul_eq_mul]
-  bilin_add_right x y z := by simp only [BilinForm.add_right, map_add]
-  bilin_smul_right r x y := by
-    simp only
-    rw [← smul_one_smul R r (_ : M), BilinForm.smul_right, smul_one_mul r (_ : R), map_smul,
-      smul_eq_mul]
+@[simps!]
+def compBilinForm (f : R →ₗ[R'] R') (B : BilinForm R M) : BilinForm R' M := LinearMap.mk₂ R'
+  (fun x y => f (B x y)) (by simp only [map_add, add_apply, forall_const])
+  (by simp only [map_smul_of_tower, smul_apply, map_smul, smul_eq_mul, forall_const])
+  (by simp only [map_add, forall_const])
+  (by simp only [map_smul_of_tower, map_smul, smul_eq_mul, forall_const])
```

Comment:
This is `restrictScalars12` combined with `compr2` I think

Reply:
Done, thanks.